pkg/machines/scripts/create_machine.sh

```diff
@@ -63,6 +65,32 @@ else
     UNATTENDED_PARAMS=""
 fi
 
+if [ "$USE_CLOUD_INIT" = "true" ]; then
+    if [ -z "$USER_PASSWORD" ]; then
+        USER_LINE=""
+        CREATE_USER_LINE=""
+    else
+        USER_LINE=$'\n'"    $USER_LOGIN:$USER_PASSWORD"
```

Comment:
The `\n` here and writing `$ROOT_LINE$USER_LINE` is error prone. I assume you do this to avoid an empty line in the list, but if you have no root password, but an user one, the result will be
```
list: |
    
    user:pwd
```
anyway, so you may as well keep it simple? Or, if the userdata really doesn't get along with empty lines, fix it harder.

Reply:
The reason I did it like this is indeed because cloud-init user-data file seems to be very sensitive to spaces and new lines.
Something like this does not work. 

```
#cloud-config
users:
  - name: test
chpasswd:
  list: |
    
    test:foobar
  expire: False
```
 
Thanks for catching this, I will repost with the fix.

Reply:
So we can't have new lines inside the list, but new lines between fields are fine. I tested the solution I am about to post with all scenarios. If you know some better templating that I can do through the script I am happy to improve it.